### ISSUE TITLE
A better shacktag interface.

### ISSUE THREAD

**Issue by AlexJWayne** [Squeegy/latest-chatty-2#81]
I hacked on an idea tonight.  I hooked text styling up to the text selection menu, and viciously killed the tags under the keyboard bullshit.

Authoring shacktags has been a pain for a while, they way it only supports adding the tags to the end, even though most of the time you want to write some text, select it, then apply a style.  Now you simply select the text, tap "Style" in the menu that appears, and choose your style in the popover modal, and your text now has that style.

And maintaining the `ComposeViewController` has been equally painful.  So much code in there is about sizing things for different screens, orientations, keyboard states.  It's a bit of a mess, and this helps clean it up.

Though it's not perfect, though I do feel it's an improvement.  On iPad the modal is huge, would be nice to make that more of an isolated popover.  And there is no cancel button, forcing you select a shacktag to apply.  I'm not sure when I'll be able to address these issues.

The code is on the `markup` branch.  So check it out, and merge into master if you deem it fit for use.  Or wait on it until it's polished up more.  Your call.
